// Function returns a promise





// const willGetYouADog = new Promise((resolve, reject) => {
// 	setTimeout(() => {
// 		const rand = Math.random();
// 		if (rand < 0.5) {
// 			resolve();
// 		}
// 		else {
// 			reject();
// 		}
// 	}, 5000);
// })



const makeDogPromise = () => {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			const rand = Math.random();
			if (rand < 0.5) {
				resolve();
			}
			else {
				reject();
			}
		}, 5000);
	});
};

// Beacause it returns, we can chain .then and .catch together, only one will run
makeDogPromise()
	.then(() => {
		console.log('YAY WE GOT A DOG!!!!');
	})
	.catch(() => {
		console.log(':( NO DOG');
	});
